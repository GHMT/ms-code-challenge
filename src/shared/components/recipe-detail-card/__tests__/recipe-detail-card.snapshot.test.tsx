import React from 'react';
import { shallow } from "enzyme";

import RecipeDetailCard from '..';

const RecipeDetailCardProps = {
  img: '',
  title: 'Dummy Title',
  description: 'A very short description',
  chefName: 'Mark Zuchiniberg',
  tags: [{ id: '1', name: 'healthy' }, { id: '2', name: 'gluten free' }]
}

describe('Recipe Detail Card Component snapshots', () => {
  it('match snapshot: all props', () => {
    const RecipeDetailCardShallowed = shallow(
      <RecipeDetailCard
        {...RecipeDetailCardProps}
      />
    );

    expect(RecipeDetailCardShallowed).toMatchSnapshot();
  });

  it('match snapshot: no chef', () => {
    const RecipeDetailCardShallowed = shallow(
      <RecipeDetailCard
        img={RecipeDetailCardProps.img}
        title={RecipeDetailCardProps.title}
        description={RecipeDetailCardProps.description}
        chefName={undefined}
        tags={RecipeDetailCardProps.tags}
      />
    );

    expect(RecipeDetailCardShallowed).toMatchSnapshot();
  });

  it('match snapshot: no tags', () => {
    const RecipeDetailCardShallowed = shallow(
      <RecipeDetailCard
        img={RecipeDetailCardProps.img}
        title={RecipeDetailCardProps.title}
        description={RecipeDetailCardProps.description}
        chefName={RecipeDetailCardProps.chefName}
        tags={undefined}
      />
    );

    expect(RecipeDetailCardShallowed).toMatchSnapshot();
  });

  it('match snapshot: no chef & no tags', () => {
    const RecipeDetailCardShallowed = shallow(
      <RecipeDetailCard
        img={RecipeDetailCardProps.img}
        title={RecipeDetailCardProps.title}
        description={RecipeDetailCardProps.description}
        chefName={undefined}
        tags={undefined}
      />
    );

    expect(RecipeDetailCardShallowed).toMatchSnapshot();
  });
});