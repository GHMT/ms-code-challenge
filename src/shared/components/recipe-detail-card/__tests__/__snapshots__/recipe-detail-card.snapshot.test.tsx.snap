// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Detail Card Component snapshots match snapshot: all props 1`] = `
<Fragment>
  <Card
    body={false}
    border="dark"
  >
    <CardHeader>
      <h2>
        Dummy Title
      </h2>
    </CardHeader>
    <CardImg
      src=""
      variant="top"
    />
    <CardBody>
      <div
        className="card-text"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          linkTarget="_blank"
          parserOptions={Object {}}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="A very short description"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
      <footer
        className="blockquote-footer"
      >
        <small
          className="text-muted"
        >
          by 
          <cite
            title="Mark Zuchiniberg"
          >
            Mark Zuchiniberg
          </cite>
        </small>
      </footer>
    </CardBody>
    <CardFooter
      className="text-muted"
    >
      <Badge
        className="mx-1 my-1 float-left"
        key="1"
        pill={false}
        variant="info"
      >
        healthy
      </Badge>
      <Badge
        className="mx-1 my-1 float-left"
        key="2"
        pill={false}
        variant="info"
      >
        gluten free
      </Badge>
    </CardFooter>
  </Card>
</Fragment>
`;

exports[`Recipe Detail Card Component snapshots match snapshot: no chef & no tags 1`] = `
<Fragment>
  <Card
    body={false}
    border="dark"
  >
    <CardHeader>
      <h2>
        Dummy Title
      </h2>
    </CardHeader>
    <CardImg
      src=""
      variant="top"
    />
    <CardBody>
      <div
        className="card-text"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          linkTarget="_blank"
          parserOptions={Object {}}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="A very short description"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Recipe Detail Card Component snapshots match snapshot: no chef 1`] = `
<Fragment>
  <Card
    body={false}
    border="dark"
  >
    <CardHeader>
      <h2>
        Dummy Title
      </h2>
    </CardHeader>
    <CardImg
      src=""
      variant="top"
    />
    <CardBody>
      <div
        className="card-text"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          linkTarget="_blank"
          parserOptions={Object {}}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="A very short description"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
    </CardBody>
    <CardFooter
      className="text-muted"
    >
      <Badge
        className="mx-1 my-1 float-left"
        key="1"
        pill={false}
        variant="info"
      >
        healthy
      </Badge>
      <Badge
        className="mx-1 my-1 float-left"
        key="2"
        pill={false}
        variant="info"
      >
        gluten free
      </Badge>
    </CardFooter>
  </Card>
</Fragment>
`;

exports[`Recipe Detail Card Component snapshots match snapshot: no tags 1`] = `
<Fragment>
  <Card
    body={false}
    border="dark"
  >
    <CardHeader>
      <h2>
        Dummy Title
      </h2>
    </CardHeader>
    <CardImg
      src=""
      variant="top"
    />
    <CardBody>
      <div
        className="card-text"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          linkTarget="_blank"
          parserOptions={Object {}}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="A very short description"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
      <footer
        className="blockquote-footer"
      >
        <small
          className="text-muted"
        >
          by 
          <cite
            title="Mark Zuchiniberg"
          >
            Mark Zuchiniberg
          </cite>
        </small>
      </footer>
    </CardBody>
  </Card>
</Fragment>
`;
